src/librustdoc/html/static/js/search.js

```diff
@@ -154,13 +389,101 @@ window.initSearch = function(rawSearchIndex) {
             return NO_TYPE_FILTER;
         }
 
-        var valLower = query.query.toLowerCase(),
-            val = valLower,
-            typeFilter = itemTypeFromName(query.type),
-            results = {}, results_in_args = {}, results_returned = {},
-            split = valLower.split("::");
+        userQuery = userQuery.trim();
+        var query = {
+            original: userQuery,
+            userQuery: userQuery.toLowerCase(),
+            length: userQuery.length,
+            pos: 0,
+            typeFilter: null,
+            elems: [],
+            elemName: null,
+            args: [],
+            returned: [],
+            // Total number of elements (includes generics).
+            totalElems: 0,
+            // Total number of "top" elements (does not include generics).
+            foundElems: 0,
```

Comment:
Looking through the code that references `foundElems`, it only checks whether `foundElems > 1`. I think it would be clearer to have a field `multi: bool` that is true when more than one element is present. The rough intuition is that it's potentially useful to know "is this a complex query or a simple one."

Reply:
It's actually tested in two places out of the parser in `innerRunQuery`. We check if there is one element or more than 1. A boolean wouldn't allow to check this.